diff -u /home/nick/Downloads/EkInspect/EkInspect.lua /media/Aurum/World of Warcraft/Interface/AddOns/EkInspect/EkInspect.lua
--- /home/nick/Downloads/EkInspect/EkInspect.lua	2011-11-28 15:53:01.000000000 -0600
+++ /media/Aurum/World of Warcraft/Interface/AddOns/EkInspect/EkInspect.lua	2012-05-24 18:40:31.718133534 -0500
@@ -46,6 +46,120 @@
 
 local MULTIBYTE_CHAR = "([\192-\255]?%a?[\128-\191]*)";
 
+local EkInspect_ResilGems = {
+	["3226"] =  4,
+	["4285"] =  25,
+	["2867"] =  2,
+	["2801"] =  8,
+	["3530"] =  20,
+	["4000"] =  40,
+	["3084"] =  5,
+	["4249"] =  25,
+	["4283"] =  25,
+	["4313"] =  25,
+	["4294"] =  25,
+	["4236"] =  20,
+	["3852"] =  15,
+	["3385"] =  12,
+	["3004"] =  20,
+	["3245"] =  20,
+	["4186"] =  30,
+	["3000"] =  12,
+	["4686"] =  25,
+	["4685"] =  25,
+	["3480"] =  8,
+	["3688"] =  7,
+	["3215"] =  12,
+	["4244"] =  20,
+	["4029"] =  20,
+	["3794"] =  15,
+	["4250"] =  25,
+	["4245"] =  35,
+	["4316"] =  25,
+	["3493"] =  8,
+	["3079"] =  4,
+	["2788"] =  9,
+	["2859"] =  3,
+	["4246"] =  35,
+	["3416"] =  6,
+	["3087"] =  4,
+	["4683"] =  25,
+	["4185"] =  20,
+	["4678"] =  25,
+	["3556"] =  10,
+	["3797"] =  20,
+	["4077"] =  40,
+	["4238"] =  20,
+	["4239"] =  208,
+	["3692"] =  7,
+	["4674"] =  25,
+	["3853"] =  40,
+	["3842"] =  25,
+	["3069"] =  4,
+	["3489"] =  8,
+	["3576"] =  10,
+	["2878"] =  4,
+	["3552"] =  10,
+	["3796"] =  20,
+	["3424"] =  6,
+	["3080"] =  5,
+	["4684"] =  25,
+	["2891"] =  10,
+	["3508"] =  8,
+	["3946"] =  35,
+	["3484"] =  8,
+	["4312"] =  25,
+	["2789"] =  15,
+	["3684"] =  7,
+	["3662"] =  14,
+	["3284"] =  5,
+	["4284"] =  25,
+	["4237"] =  20,
+	["2862"] =  3,
+	["3513"] =  8,
+	["2933"] =  15,
+	["3436"] =  6,
+	["4184"] =  10,
+	["3821"] =  8,
+	["3586"] =  10,
+	["3793"] =  15,
+	["4338"] =  50,
+	["4314"] =  25,
+	["3561"] =  10,
+	["4676"] =  50,
+	["3744"] =  34,
+	["3899"] =  30,
+	["4248"] =  25,
+	["3090"] =  4,
+	["3795"] =  20,
+	["3462"] =  16,
+	["3441"] =  6,
+	["3431"] =  6,
+	["3930"] =  15,
+	["3566"] =  10,
+	["4681"] =  25,
+	["3707"] =  7,
+	["4680"] =  25,
+	["3708"] =  7,
+	["2759"] =  8,
+	["3503"] =  8,
+	["4672"] =  25,
+	["3581"] =  10,
+	["4247"] =  35,
+	["3792"] =  20,
+	["3713"] =  7,
+	["4315"] =  25,
+	["3131"] =  10,
+	["2856"] =  4,
+	["3412"] =  6,
+	["3408"] =  6,
+	["4036"] =  67,
+	["3600"] =  6,
+	["3680"] =  7,
+	["3961"] =  18,
+	["3229"] =  12,
+};
+
 local function strlen_utf8(str)
 	-- Get length of string in characters. This may not equal the number of bytes in the string.
 	return strlenutf8(str);
@@ -672,7 +786,40 @@
 		local slot = slotNames.slot;
 		local link = GetInventoryItemLink(unit, slot);
 		if (link) then
+    
 			local itemName, link, quality, itemLevel = GetItemInfo(link);
+      
+      local _, _, ench, gem1, gem2, gem3, gem4 = strsplit(":", link);
+      
+      ench = tostring(ench);
+      gem1 = tostring(gem1);
+      gem2 = tostring(gem2);
+      gem3 = tostring(gem3);
+      gem4 = tostring(gem4);
+      
+      local itemStats = GetItemStats(link);
+      
+      local itemResilience = (itemStats["ITEM_MOD_RESILIENCE_RATING_SHORT"] or 0)
+      
+      if EkInspect_ResilGems[ench] then
+      	itemResilience = itemResilience + EkInspect_ResilGems[ench];
+      end
+      
+      if EkInspect_ResilGems[gem1] then
+      	itemResilience = itemResilience + EkInspect_ResilGems[gem1];
+      end
+      
+      if EkInspect_ResilGems[gem2] then
+      	itemResilience = itemResilience + EkInspect_ResilGems[gem2];
+      end
+      
+      if EkInspect_ResilGems[gem3] then
+      	itemResilience = itemResilience + EkInspect_ResilGems[gem3];
+      end
+      
+      if EkInspect_ResilGems[gem4] then
+      	itemResilience = itemResilience + EkInspect_ResilGems[gem4];
+      end
 
 			-- Adjust item level based on quality of the item.
 			local adjustedLevel = itemLevel;
@@ -692,7 +839,7 @@
 			count = count + 1;
 			total = total + itemLevel;
 
-			tinsert(detailsList, { item = itemName, level = itemLevel, ilevel = itemLevel, adjusted = adjustedLevel, link = link, id = id, slot = slot, quality = quality });
+			tinsert(detailsList, { item = itemName, level = itemLevel, ilevel = itemLevel, adjusted = adjustedLevel, link = link, id = id, slot = slot, quality = quality, resilience = itemResilience });
 		end
 	end
 
@@ -724,7 +871,7 @@
 				if (EkInspect_UI_Adjust[EkInspect_WhoWasInspected] == 1) then
 					detailTable.level = detailTable.adjusted;
 				else
-					detailTable.level = detailTable.ilevel;
+					detailTable.level = detailTable.resilience;
 				end
 
 				total = total + detailTable.level;
